src/main.rs

```diff
@@ -113,16 +116,35 @@ mod filters {
             .and(warp::query::<GetQuery>())
             .and(with_db(db))
             .and_then(move |query: GetQuery, db: Database| async move {
-                let filter_options = mongodb::options::FindOneOptions::builder().build();
-                let filter = bson::doc! { "path": query.path };
 
-                match db.collection("data").find_one(filter, filter_options).await {
-                    Ok(Some(doc)) => {
-                        let data: Data = bson::from_document(doc).unwrap();
-                        Ok(warp::reply::json(&data))
+                if let Some(_) = query.skip {
+                    // If there is a skip query param, retrieve collection
+
+                    let filter_options = mongodb::options::FindOptions::builder().skip(query.skip).limit(query.page_size).build();
```

Comment:
Let's add a bounds check on page_size so people can't request absurd amounts of data

Reply:
Added a limit of 100.  returns 400 of page_size if above limit.  

Also updated the Response object to have a `results` field which has the array of results, instead of just returning the array of results as the response.